hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/WritableScheduleStoreImpl.java

```diff
@@ -161,16 +160,18 @@ private Schedule markDeleted(final Schedule schedule, final Instant consensusTim
     public void purgeExpiredSchedulesBetween(long firstSecondToExpire, long lastSecondToExpire) {
         for (long i = firstSecondToExpire; i <= lastSecondToExpire; i++) {
             final var second = new ProtoLong(i);
-            final var scheduleList = schedulesByExpirationMutable.get(second);
-            if (scheduleList != null) {
-                for (final var schedule : scheduleList.schedules()) {
-                    schedulesByIdMutable.remove(schedule.scheduleIdOrThrow());
-
-                    final ProtoBytes hash = new ProtoBytes(ScheduleStoreUtility.calculateBytesHash(schedule));
-                    schedulesByEqualityMutable.remove(hash);
-                    logger.info("Purging expired schedule {} from state.", schedule.scheduleIdOrThrow());
+            final var scheduleIdList = scheduleIdsByExpirationMutable.get(second);
+            if (scheduleIdList != null) {
+                for (final var scheduleId : scheduleIdList.scheduleIds()) {
+                    final var schedule = schedulesByIdMutable.get(scheduleId);
+                    if (schedule != null) {
+                        final ProtoBytes hash = new ProtoBytes(ScheduleStoreUtility.calculateBytesHash(schedule));
+                        scheduleIdByEqualityMutable.remove(hash);
+                    }
+                    schedulesByIdMutable.remove(scheduleId);
+                    logger.info("Purging expired schedule {} from state.", scheduleId);
```

Comment:
We should probably either remove this or make it a `DEBUG` level event.

Reply:
Changed to debug level log.